src/Module.Server/Rewards/CrpgRewardServer.cs

```diff
@@ -429,15 +603,19 @@ private IList<CrpgUserDamagedItem> BreakItems(CrpgPeer crpgPeer, float roundDura
                 continue;
             }
 
+            int netCompensationForCrpgUser = netCompensationByCrpgUserId.TryGetValue(crpgPeer.User.Id, out int compensationForCrpgUser) ? compensationForCrpgUser : 0;
+            updateResult.EffectiveReward.Gold -= netCompensationForCrpgUser;

```

Comment:
Something is weird here. The backend returns how much gold was given to the player. How come we are touching that value  🤔

Reply:
Because in line 438 i combine the reward with the compensation, to only send one userUpdate per player.
I avoided sending 2 separate userUpdates because of this problem i faced while testing it
![grafik](https://github.com/verdie-g/crpg/assets/35863824/7d57ee89-3761-4755-82b8-267c263a2340)
This would require weird logic in CrpgRewardClient to split the reward message from the compensation message.

And i asked in general chat and staff how they would interpret this message, and i think all of them said that they would expect that the compensation is extra to the reward.
![grafik](https://github.com/verdie-g/crpg/assets/35863824/22a37a90-5dd7-451d-a32f-c9d075536345)

So for clarity i subtracted the compensation from the reward again for the message.